app/ide-desktop/lib/dashboard/src/utilities/object.ts

```diff
@@ -49,6 +49,26 @@ export function unsafeEntries<T extends object>(
   return Object.entries(object)
 }
 
+// ==================
+// === mapEntries ===
+// ==================
+
+/** Return the entries of an object. UNSAFE only when it is possible for an object to have
+ * extra keys. */
```

Comment:
This is my lack of TypeScript experience showing, but what does unsafe mean in this context?

Reply:
unsafe meaning incorrect here. in general most kinds of reflection (e.g. iterating over keys) are technically unsafe due to subtyping (`class Bar extends Foo` may have extra keys, yet it is still a valid `Foo` due to the liskov substitution principle)